crates/mun_codegen/src/snapshots/test__issue_225.snap

```diff
@@ -0,0 +1,75 @@
+---
+source: crates/mun_codegen/src/test.rs
+expression: "struct Num {\n    value: i64,\n}\n\npub fn foo(b: i64) {\n    Num { value: b }.value;\n}\n\npub fn bar(b: i64) {\n    { let a = Num { value: b }; a}.value;\n}"
+---
+; == FILE IR =====================================
+; ModuleID = 'main.mun'
+source_filename = "main.mun"
+
+%DispatchTable = type { i8** (i8*, i8*)* }
+%struct.MunTypeInfo = type { [16 x i8], i8*, i32, i8, i8 }
+%Num = type { i64 }
+
+@allocatorHandle = external global i8*
+@dispatchTable = external global %DispatchTable
+@global_type_table = external global [5 x %struct.MunTypeInfo*]
+
+define void @foo(i64) {
+body:
+  %init = insertvalue %Num undef, i64 %0, 0
+  %new_ptr = load i8** (i8*, i8*)*, i8** (i8*, i8*)** getelementptr inbounds (%DispatchTable, %DispatchTable* @dispatchTable, i32 0, i32 0)
+  %Num_ptr = load %struct.MunTypeInfo*, %struct.MunTypeInfo** getelementptr inbounds ([5 x %struct.MunTypeInfo*], [5 x %struct.MunTypeInfo*]* @global_type_table, i32 0, i32 2)
+  %type_info_ptr_to_i8_ptr = bitcast %struct.MunTypeInfo* %Num_ptr to i8*
+  %allocator_handle = load i8*, i8** @allocatorHandle
+  %new = call i8** %new_ptr(i8* %type_info_ptr_to_i8_ptr, i8* %allocator_handle)
+  %Num_ptr_ptr = bitcast i8** %new to %Num**
+  %Num_mem_ptr = load %Num*, %Num** %Num_ptr_ptr
+  store %Num %init, %Num* %Num_mem_ptr
+  %mem_ptr = load %Num*, %Num** %Num_ptr_ptr
```

Comment:
**Future Work**
This is a great example of where we can optimise a lot by using a `struct(value)` instead of a `struct(gc)`. For temporaries (or scoped `struct` instances) it doesn't  really make sense to allocate them on the heap.

Reply:
That would be great! But the use case is quite small, it's only when you construct a GC struct that doesn't leave the scope of the function at all (otherwise you can't make guarantees). Nonethess, if we can catch that in the compiler and turn it into a stack allocation, that would be great!